File: ArrayStrings/stock.py
class Solution(object):
    def maxProfit(self, prices):
       buy = 0
       maxprofit = 0
       for sell in range(1, len(prices)):
        
           if(prices[buy]>prices[sell]):
              buy = sell
           elif(prices[sell]>prices[buy]):
               profit = prices[sell] - prices[buy]
               sell = sell + 1
               
               if(profit>maxprofit):
                   maxprofit = profit
                   
                
       return maxprofit

File: ArrayStrings/stock.js
//? BEST TIME TO BUY AND SELL STOCK RETURN MAX PROFIT
var maxProfit = function (prices) {
  let mapOriginalIndex = {};
  for (let i = 0; i < prices.length; i++) {
    mapOriginalIndex[prices[i]] = i;
  }
  prices = prices.sort((a, b) => a - b);
  let buy = 0;
  let sell = prices.length - 1;
  let maxProfit = 0;
  console.log(mapOriginalIndex);
  for (let i = 0; i < prices.length; i++) {
    if (mapOriginalIndex[prices[sell]] > mapOriginalIndex[prices[buy]]) {
      let profit = prices[sell] - prices[buy];
      sell--;

      if (profit > maxProfit) maxProfit = profit;
    } else {
      buy++;
    }
  }
  return maxProfit;
};
let prices = [7, 1, 5, 3, 6, 4];
console.log(maxProfit(prices)); //! no work
